graph/pull_v2.go

```diff
@@ -171,24 +179,45 @@ func (p *v2Puller) download(di *downloadInfo) {
 		ID:        stringid.TruncateID(di.img.ID),
 		Action:    "Downloading",
 	})
-	io.Copy(tmpFile, reader)
+
+	downloadComplete := make(chan error)
+	readCanceler := ioutils.NewReadCloserWithError(reader)
```

Comment:
I agree, I don't see the use of this new type given that the only code path where we actually use its `CloseWithError` function could be use to set whatever error we want without having it returned by the stream itself.